tests/src/whisk/core/database/test/CouchDbRestClientTests.scala

```diff
@@ -0,0 +1,140 @@
+/*
+ * Copyright 2015-2016 IBM Corporation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package whisk.core.database.test
+
+import scala.concurrent.Await
+import scala.concurrent.Future
+import scala.concurrent.Promise
+import scala.concurrent.duration.Duration
+import scala.concurrent.duration.DurationInt
+import scala.concurrent.duration.DurationDouble
+import scala.util._
+
+import akka.actor.ActorRef
+import akka.actor.ActorSystem
+import akka.actor.Props
+import akka.http.scaladsl.model._
+
+import spray.json._
+import spray.json.DefaultJsonProtocol._
+
+import whisk.core.WhiskConfig
+import whisk.core.WhiskConfig._
+import whisk.core.database.CouchDbRestClient
+
+import whisk.test.http.RESTProxy
+
+import org.junit.runner.RunWith
+import org.scalatest.BeforeAndAfter
+import org.scalatest.BeforeAndAfterAll
+import org.scalatest.FlatSpec
+import org.scalatest.Matchers
+import org.scalatest.concurrent.PatienceConfiguration.Timeout
+import org.scalatest.concurrent.ScalaFutures
+import org.scalatest.junit.JUnitRunner
+
+@RunWith(classOf[JUnitRunner])
+class CouchDbRestClientTests extends FlatSpec
+    with Matchers
+    with BeforeAndAfterAll
+    with ScalaFutures {
+
+    implicit val actorSystem = ActorSystem()
+    implicit val executionContext = actorSystem.dispatcher
+
+    override implicit val patienceConfig = PatienceConfig(timeout = 10.seconds, interval = 0.5.seconds)
+
+    override def afterAll() {
+        Await.ready(akka.http.scaladsl.Http().shutdownAllConnectionPools(), Duration.Inf)
+        actorSystem.terminate()
+        Await.result(actorSystem.whenTerminated, Duration.Inf)
+    }
+
+    val config = new WhiskConfig(Map(
+        dbProvider -> null,
+        dbProtocol -> null,
+        dbUsername -> null,
+        dbPassword -> null,
+        dbHost -> null,
+        dbPort -> null))
+
+    val dbAuthority = Uri.Authority(
+        host = Uri.Host(config.dbHost),
+        port = config.dbPort.toInt
+    )
+
+    val proxyPort = 15975
+    val proxyActor = actorSystem.actorOf(Props(new RESTProxy("0.0.0.0", proxyPort)(dbAuthority, config.dbProtocol == "https")))
+
+    // the (fake) DB name doesn't matter as this test only execises the "/" endpoint ("instance info")
+    val client = CouchDbRestClient.make("http", "localhost", 15975, config.dbUsername, config.dbPassword, "fakeDB")
```

Comment:
woops, yes. Also, that number is completely number. Raise your hand if you have a better one.